### DISCUSSION TITLE
How to use EraseBrush in "fabric": "^6.0.0-beta9"

### DISCUSSION THREAD

**Discussion by tianmuji** [fabricjs/fabric.js#9758] [General]
I see eraser_mixins in source code
But how can I import it via fabric namespace?

I hope there is someway to import it, otherwise I have to create a Eraser_Brush in the source code

**Comment by ShaMan123** [comment]
I extracted it to https://github.com/ShaMan123/erase2d

**Reply by tianmuji** [reply]
Thank you for your great work.
I am trying to migrating it to HarmonyOS version.
Reading your source code, I get one question
1: what is the overall approach to realize eraser function ?
I find it seems to be
first you override onMouse function,
when onMouseDown triggered, you call drawEffect and parent class onMouseDown method
when onMouseMove, just call parent class onMouseMove and dispatch move event
when onMouseUp, additionally call disposer initialized in onMouseDown event, which will call erase and drawEffect method

the key here is that I do not quite understand clipping the destination context and compositing it with the source context
And why you use effectContext
Could you please explain a lit bit more in detail?
Thank you so much

**Reply by ShaMan123** [reply]
Please open a discussion in the [repo](https://github.com/ShaMan123/erase2d)

**Reply by tianmuji** [reply]
please check [https://github.com/ShaMan123/erase2d/issues/18](url)
